### ISSUE TITLE
@dikaiosune is acting like a jerk

### ISSUE THREAD

**Issue by julesmoretti** [react-community/create-react-native-app#425]
@dikaiosune is locking down issues and preventing responses from end users...

Please see: https://github.com/react-community/create-react-native-app/issues/420

Not cool dude

**Comment by dikaiosune** [comment]
Hi! I'm really sorry that I've upset you. It is certainly not my intention to cause you distress or harm. Please bear in mind when reading this that I'm writing this instead of working to triage ~50 issues that other human beings are having while trying to use this tool.

The timeline here, AFAICT, is:

## facebook/react-native#16120

You opened this issue citing a message that included `npm ERR` -- I don't see any way in which this prevented you from moving forward with your project. Were you unable to run `npm start`? Were you unable to install a dependency? This kind of information is a basic courtesy when opening a bug report. When asked to create an issue here, you responded quite rudely. Let's look at some of the specific complaints you had:

> This is because your quick start deployment guideline from facebook.github.io/react-native/docs/getting-started.html is exclusively contingent on this tool, and therefore it becomes (in my opinion) your responsibility to make sure it is maintained rather than for your users to.

React Native is free, open-source software. The maintainers are under no obligation to you, especially when the "Building your own native code" guide is also available. Yes, it is in their best interests to make sure that as many users as possible have a good experience, but as a community of software developers, it is *all* of our responsibilities to keep things maintained. Often, fixing things in open source requires navigating several bug trackers of related projects. This is the reality of how humans are able to organize to build things.

> that is one method (getting your users to do the work)

Yes, that is how open source commonly works. The fact that any of this works at all without having to have written it yourself from scratch is magical. And in many ways, it is because users have helped maintainers triage bug reports, and make sure they end up in front of the right people.

## https://github.com/react-community/create-react-native-app/issues/420

You opened this issue after being redirected from React Native, without providing any additional context, nor any explanation of why this was actually stopping you from building your project. When a repository has a new issue template, it is provided to help *you* get a solution as quickly as possible. Any time you can reduce the number of times a maintainer has to go back and forth to get the necessary information, you greatly increase your chances of getting help. This is unfortunately just the reality of a world in which there are far more people to help than people who can help them.

You happened to catch us at an unfortunate time, with one maintainer on vacation, another spending time with his newborn, and others struggling to keep up with other projects. It certainly didn't help that you just provided a link to another issue, instead of providing information about what your problem was and how it was preventing you from using the tool.

## https://github.com/facebook/react-native/issues/16174

I don't see anything different from your previous reports, and I'm still unclear what's happening. Both RN and CRNA expect you to generate projects using their scripts, and it's not surprising to me that installing the packages separately will cause problems. This, in particular, is what made it seem to me that you were spamming across several repositories instead of seeking help in good faith.

## https://github.com/expo/expo/issues/726

I also don't understand why you created this issue -- we don't support the versions of react-native and react that you installed. We tried to help you get your versions lined up, which went poorly because you were taking an unsupported path to using our tools. I was catching up through issues on this repository, and I saw effectively identical bug reports. Which is why I replied here, linked it there, and asked you to keep your reports focused on the projects where the issue is actually occurring.

## Today

I created a [comment](https://github.com/react-community/create-react-native-app/issues/420#issuecomment-334882447) explaining why these warnings appear, and also explaining that they should not prevent you from working on your project. It's unfortunate that npm has chosen to display a message including `npm ERR` for a harmless problem, but that's the way the world is today. Seeing errors about peer dependencies with `npm list` is just the unfortunate combination of npm's choices and upstream React Native's choices.

## So Far

* You haven't been prevented from using `create-react-native-app` -- the errors you're seeing are not from running the project
* Even had you been prevented from using CRNA, you would of course still have the option of running `react-native init` as it says in the React Native documentation
* You've behaved rudely and unprofessionally towards myself and maintainers of other projects
* You've provided insufficient explanations of the actual problem to get help, despite multiple requests to explain what you're trying to do
* You've now opened an issue personally attacking me. Which frankly is just not smart. I mean really, why do that? What does it gain you?

## What I'm going to do

1. Lock the issues in which you've behaved poorly, as I've already started to do.
2. Work our hardest to resolve these warnings for all users -- I agree that the current situation is terrible, and I really wish Facebook had thought more of the open source community when they chose this versioning scheme. Unfortunately, there's not a lot we can do here since while we contribute to React Native, we can't control every facet of the project.

## What I'd ask of you

1. Take a deep breath and take care of yourself -- I'm sorry that these warning/error messages have affected you personally. And I'm very sorry that it seemed like it was preventing you from being able to work on your project.
2. Try creating a fresh project, and instead of running `npm list`, try running `npm start`.
3. Don't do this again.
4. Consider how your current colleagues and potential future employers would perceive the behavior you've displayed in these threads.
